ocp_resources/data_source.py

```diff
@@ -39,19 +40,29 @@ def to_dict(self):
                 },
             })
 
-    @property
-    def pvc(self):
-        data_source_pvc = self.instance.spec.source.pvc
-        pvc_name = data_source_pvc.name
-        pvc_namespace = data_source_pvc.namespace
+    def _get_boot_source(self, boot_source_type):
+        boot_source = self.instance.spec.source.get(boot_source_type)
+        if not boot_source:
+            return None
+        boot_source_name = boot_source.name
+        boot_source_namespace = boot_source.namespace
         try:
-            return PersistentVolumeClaim(
+            boot_source_object = PersistentVolumeClaim if boot_source_type == "pvc" else VolumeSnapshot
+            return boot_source_object(
                 client=self.client,
-                name=pvc_name,
-                namespace=pvc_namespace,
+                name=boot_source_name,
+                namespace=boot_source_namespace,
             )
         except ResourceNotFoundError:
             self.logger.warning(
-                f"dataSource {self.name} is pointing to a non-existing PVC, name:"
-                f" {pvc_name}, namespace: {pvc_namespace}"
+                f"dataSource {self.name} is pointing to a non-existing {boot_source_type}, name:"
+                f" {boot_source_name}, namespace: {boot_source_namespace}"
             )
+
+    @property
+    def pvc(self):
+        return self._get_boot_source(boot_source_type="pvc")
+
+    @property
+    def snapshot(self):
+        return self._get_boot_source(boot_source_type="snapshot")
```

Comment:
`boot_source` should be the property that returns either a PVC or snapshot

Reply:
To clarify - add `boot_source` and add deprecation on `pvc`

Reply:
Just to clarify:

You are suggesting that the only property should be `boot source`,
And that it should return PersisentVolumeClaim or VolumeSnapshot dynamically based on the the instance boot source type?

Reply:
right, and add deprecation warning on `pvc` that it will be deprecated in 4.16.

Reply:
Hi

I have this PR in downstream: [https://code.engineering.redhat.com/gerrit/c/cnv-tests/+/449182](PR)
And I want to apply this DataSource change on it as well, because its accessing its snapshot/pvc references.
Its currently on main, but need to be backported to 4.15 & 4.14.

So If I change to `boot source` instead of `pvc`, the implementation will be different in 4.14,4.15.
So should I implement the downstream PR with `boot_source` and in 4.14&4.15 add the `snapshot` property like I did here?